text/0000-pack-publish-buildpack.md

```diff
@@ -0,0 +1,276 @@
+
+1. Creates a `local` branch on the local registry index cache
+1. Makes a Git commit to the local registry index with an update setting `yank=true` in the JSON payload representing the buildpack version as defined in [RFC-0022](https://github.com/buildpacks/rfcs/blob/master/text/0022-client-side-buildpack-registry.md)
+1. Uses Git to push the commit to the registry index's `origin`.
+1. Updates the `master` branch of the local registry cache from `origin`.
+
+Options:
+
+* `--undo` - this option will execute the same flow, but set `yanked=false`
+* `--auth <token>` - If the registry type would default to opening a web browser, it will instead perform the action using an API. In the case of Github, it will use the provided token to invoke the [Issues API](https://developer.github.com/v3/issues/)
+
+Example:
+
+```
+$ pack yank-buildpack buildpacks/sample@0.0.1
+```
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+* `type` of registry isn't accounted for in the [Buildpack URI](https://github.com/buildpacks/rfcs/blob/master/text/0037-buildpack-uris.md) definition
+* Github is a single-point-of-failure for the official registry
+* The user must have a Github account to publish to the official registry
+* The user may only publish to the official registry via a browser
+* "publish" is ambigious (we use it in `pack package-buildpack --publish` to mean pushing an OCI image to a Docker registry).
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Infer registry type
+
+Instead of configuring the registry `type`, Pack could infer the type from the URL. For example, any URL with the host `github.com` would be interpreted as type `github`. Any local file path would be interpreted as type `git`.
+
+## Direct commits to the remote index
+
+Instead of handling Github Issues on the `buildpacks/registry` repo, the `publish-buildpack` command could open a PR against the `buildpacks/registry-index`. This has the following advantages:
+
+* The operation could be performed headlessly without storing or transporting the user's Github token.
+
+This also has the following drawbacks:
+
+* The user *must* fork the `buildpacks/registry-index`
+* The contents of a PR to the `buildpacks/registry-index` repo are not intended to be human readable, which would make it difficult for users to audit before submitting.
+* This would result in commits against the index from the actual user, which requires that they DCO sign-off the commit (it's unclear if automating the sign-off is legally acceptable).
+* This differs from [rust-lang/crates.io-index](https://github.com/rust-lang/crates.io-index), which only has commits from a single user.
+
+## Using `push-buildpack` instead of `publish-buildpack`
+
+The command to release a new version could be `pack push-buildpack` to avoid ambiguity with other commands that use a `--publish` flag.
+
+# Prior Art
+[prior-art]: #prior-art
+
+* [rust-lang/crates.io-index](https://github.com/rust-lang/crates.io-index)
+
+# Unresolved Questions
```

Comment:
Since the `--headless` option got removed, the biggest unresolved question is how to deal with CI. This has to be a large use case like at Heroku.